src/presign/participant.rs

```diff
@@ -305,6 +270,52 @@ impl PresignParticipant {
         Ok(vec![message])
     }
 
+    fn handle_round_two_msg<R: RngCore + CryptoRng>(
+        &mut self,
+        rng: &mut R,
+        message: &Message,
+        main_storage: &Storage,
+    ) -> Result<Vec<Message>> {
+        let (auxinfo_identifier, keyshare_identifier) =
+            self.get_associated_identifiers_for_presign(&message.id())?;
+
+        // First, verify the bytes of the round two value, and then
+        // store it locally. In order to v
+        self.validate_and_store_round_two_public(
+            main_storage,
+            message,
+            auxinfo_identifier,
+            keyshare_identifier,
+        )?;
+
+        // Since we are in round 2, it should certainly be the case that all
+        // public auxinfo for other participants have been stored, since
+        // this was a requirement to proceed for round 1.
+        assert!(has_collected_all_of_others(
+            &self.other_participant_ids,
+            main_storage,
+            StorableType::AuxInfoPublic,
+            auxinfo_identifier
+        )?);
+
+        // Check if storage has all of the other participants' round two values (both
+        // private and public), and call do_round_three() if so
+        match has_collected_all_of_others(
```

Comment:
Nit: It would be clearer to me to make this an `if` statement rather than a match on a bool.

Reply:
I think there is a little too much going on this match/if expression:
- I'm not a huge fan of side effects during the "if" expression evaluation.
- There is also a couple of `?` so we have additional return paths here.

I would factor out the two function calls to `has_collected_all_of_others` like so:
```rust
        let foo = has_collected_all_of_others(
            &self.other_participant_ids,
            &self.storage,
            StorableType::RoundTwoPrivate,
            message.id(),
        )?;
        let bar = has_collected_all_of_others(
            &self.other_participant_ids,
            &self.storage,
            StorableType::RoundTwoPublic,
            message.id(),
        )?;
        if foo && bar {
            Ok(self.gen_round_three_msgs(rng, message, main_storage)?)
        } else {
            Ok(vec![])
        }
```